### PR TITLE
feat(asset): Private properties for asset sql storage

### PR DESCRIPTION
**Pull request by saschaisele-zf**
## What this PR changes/adds
This PR extends the SQL storage to accost private properties.

## Why it does that

The changes private properties bring, have to be reflected in all storage variations.

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [x] documented public classes/methods?
- [x] added/updated relevant documentation?
- [ ] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-edc/Connector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [Etiquette for pull requests](https://github.com/eclipse-edc/Connector/blob/main/pr_etiquette.md) for details_)

### CODE REVIEW

**Comment by saschaisele-zf** [comment]
Labels are still missing since i can't add them.

**Review by paullatzelsperger — COMMENTED** [review]
Regarding testing: an easy way to make sure the asset index works well with the `privateProperties` would be to always add them to generated assets:
```java
//AssetIndexTestBase.java#L630
@NotNull
protected Asset createAsset(String name, String id, String contentType) {
    return Asset.Builder.newInstance().id(id).name(name).version("1").contentType(contentType).privateProperty("test-private-prop", "test-private-val").build();
}
````

That would be sufficient, since most tests already recursively compare Assets in their assertions.

However I would like to see a test showing the behaviour when there are public and private props with the same name (name clash). IIRC we agreed to disallow this.

File: extensions/control-plane/store/sql/asset-index-sql/src/main/java/org/eclipse/edc/connector/store/sql/assetindex/schema/AssetStatements.java [hunk 1]
```diff
@@ -83,6 +83,13 @@ default String getAssetPropertyColumnType() {
         return "property_type";
     }
 
+    /**
+     * The asset property is private column.
+     */
+    default String getAssetPropertyColumnIsPrivate() {
```
[the remark this answers is not in the document]
Comment (saschaisele-zf) [reply extensions/control-plane/store/sql/asset-index-sql/src/main/java/org/eclipse/edc/connector/store/sql/assetindex/schema/AssetStatements.java]: Renamed all offenders.

File: extensions/control-plane/store/sql/asset-index-sql/src/main/java/org/eclipse/edc/connector/store/sql/assetindex/SqlPropertyWrapper.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ *  Contributors:
+ *       ZF Friedrichshafen AG - Initial Implementation
+ *
+ */
+
+package org.eclipse.edc.connector.store.sql.assetindex;
+
+import java.util.AbstractMap;
+
+public class SqlPropertyWrapper {
```
[the remark this answers is not in the document]
Comment (saschaisele-zf) [reply extensions/control-plane/store/sql/asset-index-sql/src/main/java/org/eclipse/edc/connector/store/sql/assetindex/SqlPropertyWrapper.java]: refactored it and it is now an inner class

File: extensions/control-plane/store/sql/asset-index-sql/src/test/java/org/eclipse/edc/connector/store/sql/assetindex/PostgresAssetIndexTest.java [hunk 3]
```diff
@@ -80,6 +80,16 @@ void query_byAssetProperty() {
 
     }
 
+    @Test
+    @DisplayName("Verify an asset query based on an Asset property")
+    void query_byAssetPrivateProperty() {
```
[the remark this answers is not in the document]
Comment (saschaisele-zf) [reply extensions/control-plane/store/sql/asset-index-sql/src/test/java/org/eclipse/edc/connector/store/sql/assetindex/PostgresAssetIndexTest.java:86]: changes to the `AssetIndexTestBase` have to wait until CosmosDB has been updated and the testing can be centralized for all three implementations

[the remark this answers is not in the document]
Comment (paullatzelsperger) [reply extensions/control-plane/store/sql/asset-index-sql/src/test/java/org/eclipse/edc/connector/store/sql/assetindex/PostgresAssetIndexTest.java:86]: nope, you could just override the method in `CosmosDbTest` and do a no-op

**Review by paullatzelsperger — APPROVED** [review]
looks good to me, we can do a bit of clean in a subsequent PR (e.g. the one about Cosmos)

**Pull request merged**